AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -165,26 +181,38 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
             return
         }
 
-        let syncMutationToCloudOperation =
-            SyncMutationToCloudOperation(mutationEvent: mutationEvent, api: api) { [weak self] result in
-                self?.log.verbose(
-                    "[SyncMutationToCloudOperation] mutationEvent finished: \(mutationEvent); result: \(result)")
-
-                if case .completed(let response) = result,
-                    case .failure(let error) = response {
-
-                    let processMutationErrorFromCloudOperation =
-                        ProcessMutationErrorFromCloudOperation(mutationEvent: mutationEvent,
-                                                               error: error) { [weak self] result in
-
-                        self?.log.verbose(
-                            "[ProcessMutationErrorFromCloudOperation] mutation error processed, result: \(result)")
-                        self?.completeProcessingEvent(mutationEvent)
-                    }
-                    self?.operationQueue.addOperation(processMutationErrorFromCloudOperation)
-                } else {
-                    self?.completeProcessingEvent(mutationEvent)
+        let syncMutationToCloudOperation = SyncMutationToCloudOperation(mutationEvent: mutationEvent,
+                                                                        api: api) { [weak self] result in
+            guard let self = self else {
+                return
+            }
+
+            self.log.verbose(
+                "[SyncMutationToCloudOperation] mutationEvent finished: \(mutationEvent); result: \(result)")
+
+            guard case .completed(let response) = result, case .failure(let error) = response else {
+                self.completeProcessingEvent(mutationEvent)
+                return
+            }
+
+            let processMutationErrorFromCloudOperation =
+                ProcessMutationErrorFromCloudOperation(dataStoreConfiguration: self.dataStoreConfiguration,
+                                                       mutationEvent: mutationEvent,
+                                                       api: api,
+                                                       storageAdapter: self.storageAdapter,
+                                                       error: error) { [weak self] result in
+                guard let self = self else {
+                    return
                 }
+
+                self.log.verbose("[ProcessMutationErrorFromCloudOperation] mutation error processed, result: \(result)")
+                if case let .success(mutationEventOptional) = result,
+                    let outgoingMutationEvent = mutationEventOptional {
+                    self.mutationEventPublisher.send(outgoingMutationEvent)
+                }
```

Comment:
Hmmm.. i guess it's safe to keep running if there is some error as part of this operation?

Reply:
i believe so, the operations (SyncMutationToCloudOperation/ProcessMutationErrorFromCloudOperation) should have its own retries like the retryability on API, by the time it's back here it's done all it can so i think it's better to let the system continue to operate by removing the mutationEvent and continue processing the next one